Stellar-contract.x

```diff
@@ -255,6 +258,63 @@ struct Int128Parts {
     uint64 hi;
 };
 
+struct ContractInvocation
+{
+    Hash contractID;
+    SCSymbol functionName;
+};
+
+struct AuthorizedInvocation
+{
+    ContractInvocation callStack<>;
+    SCVec topArgs;
+    uint64* nonce;
```

Comment:
What's the significance of the `nonce` being optional?

Reply:
`Vec<AuthorizedInvocation>` is basically a flattened forest of invocation trees and only the root nodes need the nonce to be consumed in order to prevent replay. This is kind of an optimization, but I think it's also useful from the UX standpoint (e.g. it allows to do multiple ops on the same token in arbitrary order as long as the token is being used by the different contracts).